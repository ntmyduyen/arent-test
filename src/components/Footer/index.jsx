import { memo } from "react"
import footer from "@data/footer"
import { Link } from "react-router-dom"
import { WrapFooter } from "./index.styles"

const Footer = () => {
    return (
        <WrapFooter>
            <div className="inner-footer">
                {footer.map(({ id, name, path }) => (
                    <Link key={id} to={path}>{name}</Link>
                ))}
            </div>
        </WrapFooter>
    )
}
export default memo(Footer)