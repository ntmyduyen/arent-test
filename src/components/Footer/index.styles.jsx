import styled from 'styled-components'

const WrapFooter = styled.footer`
    margin-top: 64px;
    height: 128px;
    background: var(--color-dark-500);
    
    .inner-footer {
        height: inherit;
        max-width: 960px;
        margin: 0 auto;
        display: flex;
        align-items: center;
        gap: 45px;

        & > * {
            text-decoration: none;
            font-style: normal;
            font-weight: 300;
            font-size: 11px;
            line-height: 16px;
            letter-spacing: 0.033px;
            color: var(--color-light);
        }
    }
`;

export {
    WrapFooter
}