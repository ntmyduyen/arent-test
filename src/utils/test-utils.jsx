/* eslint-disable import/export */
import setupStore from '@redux/index';
import { cleanup, render } from '@testing-library/react'
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { afterEach } from 'vitest'

afterEach(() => {
	cleanup()
})

function renderWithProviders(
	ui,
	{
		preloadedState = {},
		// Automatically create a store instance if no store was passed in
		store = setupStore(preloadedState),
		...renderOptions
	} = {}
) {
	function Wrapper({ children }) {
		return <Provider store={store}>
			<MemoryRouter>{children}</MemoryRouter>
		</Provider>
	}
	return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) }
}

export * from '@testing-library/react'
export { default as userEvent } from '@testing-library/user-event'
export {
	renderWithProviders
}