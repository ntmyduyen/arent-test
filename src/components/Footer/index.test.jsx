import { renderWithProviders } from "@utils/test-utils"
import Footer from "."

describe('Footer', () => { 
    it('should be match with snapshot', () => {
        const { container } = renderWithProviders(<Footer />)
        expect(container).toMatchSnapshot()
    })
})