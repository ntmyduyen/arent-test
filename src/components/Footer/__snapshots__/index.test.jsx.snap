// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer > should be match with snapshot 1`] = `
<div>
  <footer
    class="sc-beqWaB PUvnb"
  >
    <div
      class="inner-footer"
    >
      <a
        href="/"
      >
        会員登録
      </a>
      <a
        href="/"
      >
        運営会社
      </a>
      <a
        href="/"
      >
        利用規約
      </a>
      <a
        href="/"
      >
        個人情報の取扱について
      </a>
      <a
        href="/"
      >
        特定商取引法に基づく表記
      </a>
      <a
        href="/"
      >
        お問い合わせ
      </a>
    </div>
  </footer>
</div>
`;
